common/htlc.h

```diff
@@ -10,10 +10,10 @@
 #include <ccan/short_types/short_types.h>
 #include <ccan/str/str.h>
 
+#define NUM_SIDES (REMOTE + 1)
 enum side {
 	LOCAL,
 	REMOTE,
-	NUM_SIDES
```

Comment:
I believe the intent here is that this is always the last entry in this enum, so that if we ever get more than two sides to a channel (fat chance), we just insert them before the `NUM_SIDES` entry.  Standard C assigns enum values monotonically IIRC.

With `NUM_SIDES` as a `#define` you have to insert the new side to the `enum`, ***and*** update the `#define`.

But in any case there is going to be a lot more changes if we ever have more than two sides to a channel so....

Reply:
Using this comment as context for this change https://github.com/ElementsProject/lightning/pull/4146#discussion_r512341287

Reply:
The rationale for not having it as part of the `enum` but rather have a `#define` is that exhaustive matching requires us to add a `case NUM_SIDES` to all `switch` statements (and technically `if` statements) if we keep it as part of the `enum`. +1 for moving it to a `#define`.